src/components/Environment/environment-utils.ts

```diff
@@ -58,3 +95,37 @@ export const createEnvironmentKindFromReleasePlan = (
     deploymentStrategy: EnvironmentDeploymentStrategy.Manual,
   },
 });
+
+export const createCredentialsSecret = async (kubeconfig: string, namespace: string) => {
+  const resource: SecretKind = {
+    apiVersion: SecretModel.apiVersion,
+    kind: SecretModel.kind,
+    metadata: { generateName: 'env-cluster-credentials-', namespace },
+    type: 'managed-gitops.redhat.com/managed-environment',
+    stringData: { kubeconfig },
+  };
+  return k8sCreateResource({ model: SecretModel, resource });
+};
+
+export const createEnvironment = async (values: CreateEnvironmentFormValues, namespace: string) => {
+  const secret = await createCredentialsSecret(values.kubeconfig, namespace);
+  const kubeconfig = YAML.load(values.kubeconfig) as KubeConfig;
+  const resource: EnvironmentKind = {
+    apiVersion: `${EnvironmentModel.apiGroup}/${EnvironmentModel.apiVersion}`,
+    kind: EnvironmentModel.kind,
+    metadata: { name: values.name.toLowerCase().trim().split(' ').join('-'), namespace },
+    spec: {
```

Comment:
We also need to add `tags: ['managed']` to tag this as self managed environment. The tags will change based on what user selects as environment tag.

Reply:
Updated.